src/Features/LanguageServer/Protocol/Features/Diagnostics/EngineV2/DiagnosticIncrementalAnalyzer.cs

```diff
@@ -65,18 +62,6 @@ internal partial class DiagnosticIncrementalAnalyzer : IIncrementalAnalyzer
             _stateManager.ProjectAnalyzerReferenceChanged += OnProjectAnalyzerReferenceChanged;
 
             _diagnosticAnalyzerRunner = new InProcOrRemoteHostAnalyzerRunner(analyzerInfoCache, analyzerService.Listener);
-
-            GlobalOptions.AddOptionChangedHandler(this, OnGlobalOptionChanged);
-        }
-
-        private void OnGlobalOptionChanged(object? sender, OptionChangedEventArgs e)
-        {
-            if (DiagnosticAnalyzerService.IsGlobalOptionAffectingDiagnostics(e.Option) &&
-                GlobalOptions.GetOption(SolutionCrawlerRegistrationService.EnableSolutionCrawler))
-            {
-                var service = Workspace.Services.GetService<ISolutionCrawlerService>();
-                service?.Reanalyze(Workspace, this, projectIds: null, documentIds: null, highPriority: false);
```

Comment:
no more concept of having to reanalyze like this.  Note: in the LSP world, external events are already listened to, and the server (us) tells the client (vs/vscode): "you need to refresh", which starts another pull.

Reply:
Thanks for the details. Curious if/how we trigger this "you need to refresh" logic only for global options that affect diagnostics that the removed code was doing.